manager.go

```diff
@@ -0,0 +1,150 @@
+package graphql
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/GannettDigital/graphql/gqlerrors"
+	"github.com/GannettDigital/graphql/language/ast"
+)
+
+const workerStartDelay = 250 * time.Microsecond
+
+// TODO add a count of workers and a way to report that
+// TODO add a way to reclaim workers that are idle
+
+// completeRequest contains the information needed to complete a field.
+// A completeRequest is passed to a resolveManager worker which processes the request.
+// This is only used when completing multiple items concurrently such as needed for a list
+type completeRequest struct {
+	index    int
+	response chan<- completeResponse
+
+	eCtx       *executionContext
+	returnType Type
+	fieldASTs  []*ast.Field
+	info       ResolveInfo
+	value      interface{}
+}
+
+// completeResponse is the type containing the completion response which is sent over a channel as workers finish
+// processing.
+type completeResponse struct {
+	index  int
+	result interface{}
+}
+
+// resolveRequest contains the information needed to resolve a field.
+// A resolveRequest is passed to a resolveManager worker which processes the request.
+type resolveRequest struct {
+	fn       FieldResolveFn
+	name     string
+	params   ResolveParams
+	response chan<- resolverResponse
```

Comment:
I wouldn't mind adding a context with timeout here, and then adding an extra check here and there so we can stop execution if something starts to hang for whatever reason.

Reply:
The entire call is wrapped in a context https://github.com/GannettDigital/graphql/blob/master/executor.go#L87
Resolve functions are passed the context https://github.com/GannettDigital/graphql/blob/master/definition.go#L580
so assuming the resolve functions properly handle context it will be cancelled if the request times out.